language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/ExtractToFunctionCodeAction.java

```diff
@@ -445,11 +539,18 @@ private Optional<List<Symbol>> getVarAndParamSymbolsWithinRangeForExprs(LineRang
                                                                             CodeActionContext context) {
         List<Symbol> varAndParamSymbols =
                 getVisibleSymbols(context, PositionUtil.toPosition(matchedLineRange.endLine())).stream()
-                        .filter(symbol -> symbol.kind() == SymbolKind.VARIABLE || symbol.kind() == SymbolKind.PARAMETER)
-                        .filter(symbol -> context.currentSemanticModel().get().references(symbol).stream()
-                                .anyMatch(location -> PositionUtil.isRangeWithinRange(PositionUtil
-                                                .getRangeFromLineRange(location.lineRange()),
-                                        PositionUtil.toRange(matchedLineRange))))
+                        .filter(symbol -> 
+                                symbol.kind() == SymbolKind.VARIABLE || symbol.kind() == SymbolKind.PARAMETER)
+                        .filter(symbol -> 
+                                symbol.getLocation().get().lineRange().filePath().equals(matchedLineRange.filePath()))
```

Comment:
Is there a specific reason(scenario) for checking the filePath?

Reply:
We don't need this check. This will restrict the functionality of extract to function code action.